alica_engine/src/engine/TeamObserver.cpp

```diff
@@ -297,41 +302,41 @@ std::unique_ptr<SimplePlanTree> TeamObserver::sptFromMessage(AgentId agentId, co
     SimplePlanTree* curParent = nullptr;
     SimplePlanTree* cur = root.get();
 
-    for (int i = 2; i < static_cast<int>(ids.size());) {
-        const int64_t id = ids[i];
-        if (id == -1) {
-            curParent = cur;
-            cur = nullptr;
-            ++i;
-        } else if (id == -2) {
+    // start at 2, because we used the first 2 ids for creating the root node
+    for (int i = 2; i < static_cast<int>(ids.size()); ) {
+        const int64_t currentEpId = ids[i];
+        if (currentEpId == -1) {
+            // go up one level
             cur = curParent;
             if (cur == nullptr) {
-                ALICA_WARNING_MSG("TO: Malformed SptMessage from " << agentId);
-                return nullptr;
+                // went up one level from root node, finished parsing
+                break;
             }
             curParent = cur->getParent();
-            ++i;
```

Comment:
don't we still need to increment i here?

Reply:
@bjoernschroeder Can you address this one?

Reply:
@bjoernschroeder

Reply:
No I dont think we need to increment here. After -1, the next id will be the currentEPId followed by the currentStateId. If you have a list like that:

[..., ep_0, s_0, -1, ep_1, s_1]
with ep_i is the i-th entrypoint id and s_i is the i-th state id.

If the value at the current index is -1, we can just continue with the next entry in the list.